src/content/peerconnection/negotiate-timing/js/main.js

```diff
@@ -0,0 +1,232 @@
+  if (currentVideoCount < videoCount) {
+    console.log('Adding ' + (videoCount - currentVideoCount) + ' transceivers');
+    for (let i = currentVideoCount; i < videoCount; ++i) {
+      pc.addTransceiver('video');
+    }
+  } else if (currentVideoCount > videoCount) {
+    console.log('Stopping ' + (currentVideoCount - videoCount) + ' transceivers');
+    for (let i = videoCount; i < currentVideoCount; ++i) {
+      currentVideoTransceivers[i].stop();
+    }
+  } else {
+    console.log(`No adjustment, video count is ${currentVideoCount}, target was ${videoCount}`);
+  }
+}
+
+async function getAudioImpairment(audioTransceiver) {
+  const stats = await audioTransceiver.receiver.getStats();
+  let currentImpairment;
+  stats.forEach(stat => {
+    if (stat.type == 'track') {
+      currentImpairment = stat.concealedSamples;
+    }
+  });
+  console.log('Found impairment value ', currentImpairment);
+  return currentImpairment;
+}
+
+async function baselineAudioImpairment(pc) {
+  audioTransceiver = pc.getTransceivers().filter(tr => tr.receiver.track.kind == 'audio')[0];
+  console.log('Found audio transceiver');
+  audioImpairmentAtStart = await getAudioImpairment(audioTransceiver);
+}
+
+async function measureAudioImpairment(pc) {
+  const startTime = performance.now();
+  const audioImpairmentNow = await getAudioImpairment(audioTransceiver);
+  console.log('Measurement took ' + (performance.now() - startTime) + ' msec');
+  return audioImpairmentNow - audioImpairmentAtStart;
+}
+
+
+async function renegotiate() {
+  adjustTransceiverCounts(pc1, parseInt(videoSectionsField.value));
+  renegotiateButton.disabled = true;
+  await baselineAudioImpairment(pc2);
+  const previousVideoTransceiverCount = pc2.getTransceivers().filter(tr => tr.receiver.track.kind == 'video').length;
+  const startTime = performance.now();
+  const offer = await pc1.createOffer();
+  await pc1.setLocalDescription(offer);
+  await pc2.setRemoteDescription(offer);
+  const answer = await pc2.createAnswer();
+  await pc1.setRemoteDescription(answer);
+  await pc2.setLocalDescription(answer);
+  const elapsedTime = performance.now() - startTime;
+  console.log(`Renegotiate finished after ${elapsedTime} milliseconds`);
+  renegotiateButton.disabled = false;
+  const fixedTime = elapsedTime.toFixed(2);
+  const currentVideoTransceiverCount = pc2.getTransceivers().filter(tr => tr.receiver.track.kind == 'video').length;
+  const audioImpairment = await measureAudioImpairment(pc2);
+  logToScreen(`Negotiation from ${previousVideoTransceiverCount} to ${currentVideoTransceiverCount} video transceivers took ${elapsedTime.toFixed(2)} milliseconds, audio impairment ${audioImpairment}`);
+}
+
+function hangup() {
+  console.log('Ending call');
+  pc1.close();
+  pc2.close();
+  pc1 = null;
+  pc2 = null;
+
+  const videoTracks = localStream.getVideoTracks();
+  videoTracks.forEach(videoTrack => {
+    videoTrack.stop();
+    localStream.removeTrack(videoTrack);
+  });
```

Comment:
I don't understand the reason for this. It seems like if you stop all the local video tracks, I need to reload the page before I can click call again.

Reply:
The most important thing to be done here is to release the camera and microphone (in case I need to use them elsewhere).
This is copied from the previous sample, though; I haven't thought deeply about this.